### PR TITLE
Fix html input

### PR DESCRIPTION
**Pull request by emord**
@orangejenny fixes the html weirdness

cc: @NoahCarnahan

### CODE REVIEW

File: src/richText.js [hunk 1]
```diff
@@ -60,7 +60,6 @@ define([
         },
         downcast: function(element) {
             element.setHtml(applyFormats($(element.getOuterHtml()).data()));
-            element.replaceWithChildren();
```
Comment (emord) [inline src/richText.js]: this never actually did anything

File: src/richText.js [hunk 2]
```diff
@@ -103,7 +102,18 @@ define([
...
+        span: {
+            classes: 'label,label-*',
+            attributes: 'data-output-value,data-value,data-date-format',
+        },
+        i: {
+            classes: 'fcc,fcc-*,fa,fa-*',
+        },
+        output: {
+            attributes: 'value',
+        },
```
Comment (millerdev) [inline src/richText.js]: Before CKEditor we did not do anything special to markup typed in labels (which can be handy when rendering as markdown). Does this have an impact on that?

Comment (emord) [inline src/richText.js]: I'm not sure why markdown would matter here? Within ckeditor you only use plain text like `*bold*` or `**italics**` it's only rendered in html in a separate div using markdown-it

Comment (millerdev) [inline src/richText.js]: I'm not sure if we support it, but [some flavors of markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#html) support inline HTML.

File: src/richText.js [hunk 3]
```diff
@@ -484,9 +494,22 @@ define([
     }
 
     function unwrapBubbles(text) {
-        var el = $('<div>').html(text);
-        el.find('.label-datanode').children().unwrap();
-        return el.text();
+        var el = $('<div>').html(text),
+            outputMapping = {},
+            ret;
+        el.find('.label-datanode').replaceWith(function(index, output) {
```
Comment (orangejenny) [inline src/richText.js]: explanatory comment would be nice, this isn't exactly self-documenting code

**Comment by orangejenny** [comment]
Looks all right to me, leaving open because I"m not sure if @millerdev 's comments are resolved.

File: src/richText.js (hunk 3 above)
Comment (emord) [inline src/richText.js]: https://github.com/dimagi/Vellum/pull/679/commits/afeafd3a2db45897064e99495736d6c125dc1846

File: src/richText.js (hunk 2 above)
Comment (emord) [inline src/richText.js]: https://github.com/dimagi/Vellum/blob/7b6c26b3786f5e15dba7722114dc427c6ad185ac/src/markdown.js#L9 
https://confluence.dimagi.com/display/commcarepublic/Text+Formatting

we have a pretty strict whitelist on what we support. 

I also attempted to put html in a form on my phone and it hid the entire text entire text.

```text <b>bold</b>``` would output ```text ``` on a phone

**Comment by emord** [comment]
expanding on

> fixes the html weirdness

This fixes these issues:

1) when typing < or > ckeditor attempts to interpret those as html tags unless you have a space in between them so `< b` would be `&lt; b` but `<b` would start to be a tag and not be replicated to the non default language properly

2) if you had text `<b>text</b>` it would be shown in the editor as **text**

File: src/richText.js (hunk 2 above)
Comment (millerdev) [inline src/richText.js]: fair enough

**Comment by millerdev** [comment]
> expanding on
>
>> fixes the html weirdness
>
> This fixes these issues:

let the whack-a-mole games begin

**Pull request merged**